plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -7319,6 +7320,55 @@ public void testSortedWritingTempStaging()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testOptimizeSortedFileRead()
+    {
+        String tableName = "test_optimize_sorted_file_read";
+        @Language("SQL") String createTableSql = format("" +
+                        "CREATE TABLE %s " +
+                        "WITH (" +
+                        "   bucket_count = 8," +
+                        "   bucketed_by = ARRAY['orderkey']," +
+                        "   sorted_by = ARRAY['orderkey']" +
+                        ") AS " +
+                        "SELECT * FROM tpch.tiny.lineitem",
+                tableName);
+        assertUpdate(createTableSql, 60175L);
+
+        @Language("SQL") String expected = "SELECT orderkey FROM lineitem ORDER BY 1 NULLS FIRST LIMIT 100";
+        @Language("SQL") String actual = format("SELECT orderkey FROM %s ORDER BY 1 NULLS FIRST LIMIT 100", tableName);
+        assertQuery(getSession(), actual, expected, assertPartialTopNCount(1));
+
+        assertQuery(
+                Session.builder(getSession())
+                        .setCatalogSessionProperty("hive", "optimize_sorted_file_reads", "true")
+                        .build(),
+                actual,
+                expected,
+                assertPartialTopNCount(0));
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    private Consumer<Plan> assertPartialTopNCount(int expectedPartialTopNCount)
+    {
+        return plan -> {
+            int actualPartialTopNCount = searchFrom(plan.getRoot())
+                    .where(node -> node instanceof TopNNode && ((TopNNode) node).getStep() == TopNNode.Step.PARTIAL)
+                    .findAll()
+                    .size();
+            if (actualPartialTopNCount != expectedPartialTopNCount) {
```

Comment:
why not just `assertNotEquals`?

Reply:
i think we would use assertEquals, but it's simpler to construct the error string this way